core/src/Streamly/Internal/Data/Fold/Type.hs

```diff
@@ -1110,26 +1131,113 @@ lmapM f (Fold step begin done) = Fold step' begin done
     where
     step' x a = f a >>= step x
 
+-- | Postscan the input of a 'Fold' to change it in a stateful manner using
+-- another 'Fold'.
+--
+-- @postscan scanner collector@
+--
+-- /Pre-release/
+{-# INLINE postscan #-}
+postscan :: Monad m => Fold m a b -> Fold m b c -> Fold m a c
+postscan (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
+    Fold step initial extract
+
+    where
+
+    {-# INLINE runStep #-}
+    runStep actionL sR = do
+        rL <- actionL
+        case rL of
+            Done bL -> do
+                rR <- stepR sR bL
+                case rR of
+                    Partial sR1 -> Done <$> extractR sR1
+                    Done bR -> return $ Done bR
+            Partial sL -> do
+                !b <- extractL sL
+                rR <- stepR sR b
+                return
+                    $ case rR of
+                        Partial sR1 -> Partial (sL, sR1)
+                        Done bR -> Done bR
+
+    initial = do
+        r <- initialR
+        rL <- initialL
+        case r of
+            Partial sR ->
+                case rL of
```

Comment:
Move `rL <- initialL` here?

Reply:
Isn't it an extra effect of `r` is `Done`.

Reply:
This existing code, just moved in this PR. I won't change it in this PR.